src/loader.js

```diff
@@ -29,12 +32,17 @@ const findModuleById = (modules, id) => {
 
 export function pitch(request) {
   const query = loaderUtils.getOptions(this) || {};
+
+  validateOptions(schema, query, 'Mini CSS Extract Plugin Loader');
+
   const loaders = this.loaders.slice(this.loaderIndex + 1);
   this.addDependency(this.resourcePath);
   const childFilename = '*'; // eslint-disable-line no-path-concat
   const publicPath =
     typeof query.publicPath === 'string'
       ? query.publicPath
+      : typeof query.publicPath === 'function'
+      ? query.publicPath(this.resourcePath, this.rootContext)
```

Comment:
We don't need here `url` as in `file-loader` because `file-loader` for other purpose (answer on above question)

Reply:
@evilebottnawi in `file-loader` it's because `file-loader` can choose to output the file itself? Doesn't the plugin do exactly that?

Reply:
@karlvr @evilebottnawi Just wanted to let you know that this introduces a breaking change in the case when the publicPath is set to an empty string meaning that resources are located in the same directory as the css file, e.g. `dist/app.css`, `dist/aResource.png`. With the introduced change the publicPath is transformed to a `/` and so the resource path becomes `/aResource.png` which is clearly not desired.

Reply:
@evilebottnawi @rmja  

That change was to follow the same behaviour in https://github.com/webpack-contrib/file-loader/blob/master/src/index.js#L37

I'm really sorry about this regression! @evilebottnawi shall we introduce a special case for an empty string here?

Reply:
@karlvr I have created https://github.com/webpack-contrib/mini-css-extract-plugin/pull/384